From 4bbc9d7e539f3b12e6f738fb3103057c64bc8bef Mon Sep 17 00:00:00 2001
From: Vitaly Andrianov <vitalya@ti.com>
Date: Fri, 22 Jul 2016 22:58:56 +0000
Subject: [PATCH 32/34] ARM: dts: k2hk: add dts node for k2hk hw_rng driver

This patch adds dts node for k2hk hw_random generator driver

Signed-off-by: Vitaly Andrianov <vitalya@ti.com>
Signed-off-by: Jacob Stiffler <j-stiffler@ti.com>
---
 arch/arm/boot/dts/keystone-k2hk-netcp.dtsi | 20 ++++++++++++++++++++
 1 file changed, 20 insertions(+)

diff --git a/arch/arm/boot/dts/keystone-k2hk-netcp.dtsi b/arch/arm/boot/dts/keystone-k2hk-netcp.dtsi
index 145ae47..e0f987a 100644
--- a/arch/arm/boot/dts/keystone-k2hk-netcp.dtsi
+++ b/arch/arm/boot/dts/keystone-k2hk-netcp.dtsi
@@ -638,3 +638,23 @@ netcpx: netcp@2f00000 {
 		};
 	};
 };
+
+sa_subsys: subsys@20c0000 {
+	#address-cells = <1>;
+	#size-cells = <1>;
+	ranges = <0 0x20c0000 0x40000>;
+	compatible = "simple-bus";
+
+	sa_config: subsys {
+		compatible      = "syscon";
+		reg             = <0x0 0x100>;
+	};
+
+	hwrng {
+		compatible = "ti,keystone-rng";
+		sa-cfg = <&sa_config>;
+		clocks = <&clksa>;
+		clock-names = "fck";
+		reg = <0x24000 0x1000>;
+	};
+};
-- 
1.9.1

